builtin/logical/database/path_roles.go

```diff
@@ -2,263 +2,606 @@ package database
+	entry, err := logical.StorageEntryJSON(databaseRolePath+name, role)
+	if err != nil {
+		return nil, err
+	}
 
-		return &logical.Response{
-			Data: data,
-		}, nil
+	if err := req.Storage.Put(ctx, entry); err != nil {
+		return nil, err
 	}
+
+	return nil, nil
 }
 
-func (b *databaseBackend) pathRoleList() framework.OperationFunc {
-	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-		entries, err := req.Storage.List(ctx, "role/")
-		if err != nil {
-			return nil, err
+func (b *databaseBackend) pathStaticRoleCreateUpdate(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	name := data.Get("name").(string)
+	if name == "" {
+		return logical.ErrorResponse("empty role name attribute given"), nil
+	}
+
+	role, err := b.StaticRole(ctx, req.Storage, data.Get("name").(string))
+	if err != nil {
+		return nil, err
+	}
+
+	if role == nil {
+		role = &roleEntry{}
+	}
+
+	if err := pathRoleCreateUpdateCommon(ctx, role, req.Operation, data); err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	// createRole is a boolean to indicate if this is a new role creation. This
+	// is used to ensure we do not allow an existing role to be "migrated" to
+	// role with a static account. If createRole is false and static_account
+	// data is given, return an error
+	createRole := req.Operation == logical.CreateOperation
+	if role == nil {
+		role = &roleEntry{}
+		createRole = true
+	}
+
+	// Static Account information
+	// If we're updating and there is no static account, error. If we're
+	// creating, add an empty static account
+	if role.StaticAccount == nil {
+		if !createRole {
+			return logical.ErrorResponse("cannot change an existing role to a static account"), nil
 		}
+		role.StaticAccount = &staticAccount{}
+	}
 
-		return logical.ListResponse(entries), nil
+	username := data.Get("username").(string)
+	if username == "" && createRole {
+		return logical.ErrorResponse("username is a required field to create a static account"), nil
 	}
-}
 
-func (b *databaseBackend) pathRoleCreateUpdate() framework.OperationFunc {
-	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-		name := data.Get("name").(string)
-		if name == "" {
-			return logical.ErrorResponse("empty role name attribute given"), nil
+	if role.StaticAccount.Username != "" && role.StaticAccount.Username != username {
+		return logical.ErrorResponse("cannot update static account username"), nil
+	}
+	role.StaticAccount.Username = username
+
+	// If it's a Create operation, both username and rotation_period must be included
+	rotationPeriodSecondsRaw, ok := data.GetOk("rotation_period")
+	if !ok && createRole {
+		return logical.ErrorResponse("rotation_period is required to create static accounts"), nil
+	}
+	if ok {
+		rotationPeriodSeconds := rotationPeriodSecondsRaw.(int)
+		if rotationPeriodSeconds < 5 {
```

Comment:
It would be good if this was explicitly related to `queueTickInterval`. Maybe make a constant set to 5 that can be referenced in both the `queueTickInterval` definition and here.